src/factories/cap.ts

```diff
@@ -0,0 +1,58 @@
+import { useSubscription } from '@prefecthq/vue-compositions'
+import { Graphics, Rectangle, Sprite, Texture } from 'pixi.js'
+import { waitForApplication } from '@/objects'
+
+type CapStyle = {
+  height: number,
+  radius: number,
+}
+
+async function cap({ height, radius }: CapStyle): Promise<Texture> {
+  const application = await waitForApplication()
+
+  const graphic = new Graphics()
+  graphic.beginFill('#fff')
+  graphic.drawRoundedRect(0, 0, radius * 2, height, radius)
+  graphic.endFill()
+
+  const cap = application.renderer.generateTexture(graphic, {
+    // drew a rounded rectangle and then just using half of the graphic to get just the left "cap"
+    region: new Rectangle(0, 0, radius, height),
+
+    // manually bumping up the resolution to keep the border radius from being blurry
+    resolution: 10,
```

Comment:
This is a little sus to me, like the global resolution should make it so that things render without dithering unless zoomed in to an unusable degree (the live version clamps the zoom in). I believe this also makes your texture a lot larger, but probably not big deal.

Reply:
Correct it creates a larger texture. 300px tall for a 30px cap. But I found even using the default resolution it immediately got blurry when zooming. I believe because it's creating the texture at the exact dimensions requested. So any rounded edges are going to pixilate as soon as you zoom even the tiniest bit. Increasing the resolution means packing in more pixels and gets much cleaner edges. We can play with just how high of a resolution is needed. But I believe it must be higher than the default in order to zoom at all.